src/server/src/routes/student/index.ts

```diff
@@ -142,6 +143,68 @@ router.get(
     }
 );
 
+/******************************************************************************
+ *            GET Request - Search - /api/student/search"
+ ******************************************************************************/
+
+interface StudentSearchRequest extends Request {
+    query: {
+        firstName?: string;
+        lastName?: string;
+        email?: string;
+        sid?: string;
+        departmentIds?: string[];
+        classStandings?: classStandings[];
+        page: string;
+        numOfItems: string;
+    };
+}
+
+router.get(
+    '/search',
+    passport.authenticate('jwt', { session: false }),
+    validationMiddleware({ querySchema: studentSearchSchema }),
+    async (req: StudentSearchRequest, res: Response) => {
+        const { departmentIds, page, numOfItems } = req.query;
+        let { firstName, lastName, email, sid, classStandings } = req.query;
+
+        if (!firstName) firstName = "";
+        if (!lastName) lastName = "";
+        if (!email) email = "";
+        if (!sid) sid = "";
+        var departmentIdInts = [];
+        // Pass -1 when the input is empty because it causes a sql parse error
+        // when we pass in an empty array.
+        if (!departmentIds) departmentIdInts = [-1];
+        else {
+            for (const id of departmentIds) {
```

Comment:
Nothing wrong here. but using the .map function is a best practice. Like this:

`departmentIdInts = departmentIds.map((id) => parseInt(id, 10));`

I think you also need to pass 10 to tell the parse function you are doing a base 10 conversion.

Reply:
Done.